### ISSUE TITLE
Need more Get function in the Cache class

### ISSUE THREAD

**Issue by PrimeObjects** [aspnet/Caching#91]
Hi,

In session class, we have Get, GetInt32, GetString functions.
Can I suggest we also do it to Cache Class to make the design consistent?

Thanks
Gary

**Comment by PrimeObjects** [comment]
Also can I suggest to have Set, SetString, SetInt32 ...

**Comment by shanselman** [comment]
Yes, we need to be able to do "StringSet" for example, if Redis is underneath IDistributedCache...
